main.go

```diff
@@ -40,6 +42,21 @@ func init() {
 	}()
 }
 
+func GenerateRandomString(length int) string {
+	s := rand.NewSource(time.Now().UnixNano())
+	r := rand.New(s)
```

Comment:
Rename `r` here to `generator` or something similar. When `r` is redeclared on L50,  it isn't clear what its type should be. Renaming the instance of `Rand` solves that.

Reply:
you're so right - completely didn't spot that!

I've renamed both variables to clear things up even more!